File: testdata/bench/javascript/BenchmarkTest00001.js
// Vulnerable: SQL injection via string concatenation
const express = require('express');
const db = require('./db');
const router = express.Router();
router.get('/test001', (req, res) => {
    const param = req.query.input;
    const sql = "SELECT * FROM users WHERE id = " + param;
    db.query(sql, (err, results) => { res.json(results); });
});
module.exports = router;

File: testdata/bench/rust/BenchmarkTest00001.rs
use actix_web::{web, HttpResponse};
use std::collections::HashMap;

// Vulnerable: SQL injection via format! macro
async fn handler(query: web::Query<HashMap<String, String>>) -> HttpResponse {
    let param = query.get("input").unwrap_or(&String::new()).clone();
    let sql = format!("SELECT * FROM users WHERE id = {}", param);
    // sqlx::query(&sql).execute(&pool).await
    HttpResponse::Ok().body(sql)
}
